src/linear_algebra/multilinear.lean

```diff
@@ -1042,3 +1042,40 @@ def multilinear_curry_right_equiv :
   right_inv := multilinear_map.uncurry_curry_right }
 
 end currying
+
+section submodule
+
+variables {R M M₂}
+[ring R] [∀i, add_comm_monoid (M₁ i)] [add_comm_monoid M'] [add_comm_monoid M₂]
+[∀i, semimodule R (M₁ i)] [semimodule R M'] [semimodule R M₂]
+
+namespace multilinear_map
+
+/-- The pushforward of an indexed collection of submodule `p i ⊆ M₁ i` by `f : M₁ → M₂`.
+
+Note that this is not a submodule - it is not closed under addition. -/
+def map [nonempty ι] (f : multilinear_map R M₁ M₂) (p : Π i, submodule R (M₁ i)) :
+  sub_mul_action R M₂ :=
+{ carrier   := f '' { v | ∀ i, v i ∈ p i},
+  smul_mem' := by {
+    rintro c _ ⟨x, hx, rfl⟩,
+    tactic.unfreeze_local_instances,
+    obtain ⟨i⟩ := ‹nonempty ι›,
```

Comment:
```suggestion
    let i : ι := nonempty.some ‹nonempty ι›,
```
You don't need to destroy the instance to obtain a point.

Reply:
But nor do I need the axiom of choice in this proof - using it seems overkill

Reply:
Honestly, I don't really mind, but the general philosophy in mathlib is that the axiom of choice is not harmful at all: it's just a (true) fact, so if avoiding it makes a proof just a little bit longer or less transparent then it's better to use it. 

Still, if you prefer to merge your version, I'm fine with it. I have already `bors d+`-ed (I'm not sure that's really a verb, but anyway), so do as you prefer.

Reply:
Since this is a proof, you can also use `by inhabit ι; exact ... default ι ...`.  In general, please don't use `tactic.unfreeze_local_instances`.  Use `unfreezing` or its interactive version `unfreezingI` instead.

Reply:
I've since modified the proof to not use `unfreeze_local_instances`. I wasn't aware of `inhabit`, that's a useful one to know.

The only reason I used `tactic.unfreeze_local_instances` is because the error message told me to!